mp/src/game/client/c_particle_system.cpp

```diff
@@ -90,6 +94,8 @@ void C_ParticleSystem::PostDataUpdate( DataUpdateType_t updateType )
 {
 	BaseClass::PostDataUpdate( updateType );
 
+	m_pEffectEntity = m_bAttachToPlayer ? (CBaseEntity*)UTIL_PlayerByIndex(1) : this;
```

Comment:
The local player is an entity, so the cast isn't needed.

Reply:
The compiler throws an error otherwise.